modules/core/src/directx.cpp

```diff
@@ -260,35 +264,35 @@ Context& initializeContextFromD3D11Device(ID3D11Device* pD3D11Device)
     if (status != CL_SUCCESS)
         CV_Error(cv::Error::OpenCLInitError, "OpenCL: Can't get number of platforms");
 
-    // TODO Filter platforms by name from OPENCV_OPENCL_DEVICE
-
     int found = -1;
     cl_device_id device = NULL;
     cl_uint numDevices = 0;
     cl_context context = NULL;
 
-    // try with CL_PREFERRED_DEVICES_FOR_D3D11_KHR
+#ifdef HAVE_CLNVEXT
+    // try with CL_PREFERRED_DEVICES_FOR_D3D11_NV(Need to check it first)
     for (int i = 0; i < (int)numPlatforms; i++)
     {
-        clGetDeviceIDsFromD3D11KHR_fn clGetDeviceIDsFromD3D11KHR = (clGetDeviceIDsFromD3D11KHR_fn)
-                clGetExtensionFunctionAddressForPlatform(platforms[i], "clGetDeviceIDsFromD3D11KHR");
-        if (!clGetDeviceIDsFromD3D11KHR)
```

Comment:
Non-NVIDIA code should be restored.

Reply:
There's no problem. Both codes are there.